File: src/main/java/swf/army/mil/blog/post/PostService.java
package swf.army.mil.blog.post;

import jakarta.persistence.EntityNotFoundException;
import jakarta.transaction.Transactional;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class PostService {

    private final PostRepository postRepository;

    public PostService(PostRepository r) { this.postRepository = r; }

    @Transactional
    public Post save(Post p) {
        return postRepository.save(p);
    }

    public Post findByID(Long id) {
        return postRepository.findById(id)
                .orElseThrow(() -> new EntityNotFoundException("Post not found"));
    }

    public List<Post> findAll() {
        return postRepository.findAll();
    }

    @Transactional
    public void delete(Long id) {
        Post p = findByID(id);

        postRepository.delete(p);
    }

    @Transactional
    public Post updateContent(Long id, String content) {
        Post p = findByID(id);

        p.setContent(content);
        return postRepository.save(p);
    }

    @Transactional
    public Post like(Long id) {
        Post p = findByID(id);

        p.setLikes(p.getLikes() + 1);
        return postRepository.save(p);
    }
}

File: src/main/kotlin/com/personal/blog/post/PostService.kt
package com.personal.blog.post

import org.springframework.stereotype.Service

@Service
class PostService(val repo: PostRepository) {

    fun create(post: CreatePostRequest): Post =
        repo.save(Post(
            title = post.title,
            content = post.content))

    fun retrieve(): List<Post> = repo.findAll()

    fun retrieve(id: Long): Post {
        val myPost = repo.findById(id)
            .orElseThrow { NoSuchElementException("Post not found") }

        return myPost
    }

    fun update(post: UpdatePostRequest): Post {
        val existingPost = repo.findById(post.id)
            .orElseThrow { NoSuchElementException("Post not found") }

        existingPost.title = post.title
        existingPost.content = post.content

        return repo.save(existingPost)
    }

    fun delete(id: Long) {
        val existingPost = repo.findById(id)
            .orElseThrow { NoSuchElementException("Post not found") }

        repo.delete(existingPost)
    }
}